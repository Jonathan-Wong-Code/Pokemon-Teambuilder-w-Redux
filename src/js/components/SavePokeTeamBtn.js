import React from 'react';

const SavePokeTeamBtn = ({ toggleModal, type, pokeTeam }) => {
  const onButtonClick = () => {
    toggleModal();
  };

  const disabledStatus = pokeTeam.length < 6 ? true : false;

  return (
    <div className='save-team' data-test='save-team-btn-component'>
      {pokeTeam.length !== 0 && (
        <button 
          onClick={onButtonClick} 
          className='save-team__btn btn' 
          disabled={disabledStatus}
          data-test='save-team-submit'
        >
          {type === 'create' ? 'Save Team' : 'Edit Team'}
        </button>
      )}
      <p className='save-team__text'> Search and pick 6 Pokemon to save. 
        { window.innerWidth > 550 &&  
          (
            <span data-test='save-team-current-num'>
              Currently {pokeTeam.length}/6.
            </span>
          )
        } 
      </p>
      {window.innerWidth < 550 && <p className='save-team__text'>Currently {pokeTeam.length}/6.</p>}
    </div>
  );
};

export default SavePokeTeamBtn;
