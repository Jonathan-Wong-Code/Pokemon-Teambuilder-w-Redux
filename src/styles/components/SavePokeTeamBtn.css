.save-team {
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.save-team__btn { 
  text-transform: uppercase;
  padding: 1rem 2rem;
  margin: 3rem auto 0 auto;
}





