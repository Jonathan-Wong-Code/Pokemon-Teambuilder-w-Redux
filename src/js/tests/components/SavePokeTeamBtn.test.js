import React from 'react';
import { shallow } from 'enzyme';
import SavePokeTeamBtn from './../../components/SavePokeTeamBtn';
import pokemon from './../fixtures/pokemon';

describe('The SavePokeTeamBtn component with type of create and full team', () => {
  let wrapper, type, toggleModal, pokeTeam;
  beforeEach(() => {
    type = 'create';
    toggleModal = jest.fn();
    pokeTeam = [pokemon, pokemon, pokemon, pokemon, pokemon, pokemon];

    const defaultProps = {
      type,
      toggleModal,
      pokeTeam
    };

    wrapper = shallow(<SavePokeTeamBtn {...defaultProps} />);
  });

  it('Should render a snapshot of the component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render the component', () => {
    expect(wrapper.find(`[data-test='save-team-btn-component']`).length).toBe(1);
  });

  it('Should render the button when pokemon selected', () => {
    expect(wrapper.find(`[data-test='save-team-submit']`).length).toBe(1);
  });

  it('Should have a button that is not disabled', () => {
    expect(wrapper.find(`[data-test='save-team-submit']`).prop('disabled')).toBe(false);
  });

  it('Should be disabled if the team length is less than 6', () => {
    wrapper.setProps({ 
      pokeTeam : [pokemon]
    });

    expect(wrapper.find(`[data-test='save-team-submit']`).prop('disabled')).toBe(true);
  });

  it(`Should have a button with a text of 'Save Team' for type = 'create'`, () => {
    expect(wrapper.find(`[data-test='save-team-submit']`).text()).toBe('Save Team');
  });

  it(`Should have a button with a text of 'Edit Team' for type !== 'create'`, () => {
    wrapper.setProps({
      type : 'edit'
    });

    expect(wrapper.find(`[data-test='save-team-submit']`).text()).toBe('Edit Team');
  });

  it('Should open the Save Team modal when the button is clicked with a full team', () => {
    wrapper.find(`[data-test='save-team-submit']`).simulate('click');
    expect(toggleModal).toHaveBeenCalled();
  });

  it('Should render the correct number of chosen pokemon in the span', () => {
    expect(wrapper.find(`[data-test='save-team-current-num']`).text()).toContain('6/6');
  });

  it('Should not render the button when there are no pokemon in the team', () => {
    wrapper.setProps({
      pokeTeam : []
    });

    expect(wrapper.find(`[data-test='save-team-submit']`).length).toBe(0);
  });
});
