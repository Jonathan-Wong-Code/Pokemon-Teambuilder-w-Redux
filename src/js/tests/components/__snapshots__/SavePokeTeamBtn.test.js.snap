// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The SavePokeTeamBtn component with type of create and full team Should render a snapshot of the component 1`] = `
<div
  className="save-team"
  data-test="save-team-btn-component"
>
  <button
    className="save-team__btn btn"
    data-test="save-team-submit"
    disabled={false}
    onClick={[Function]}
  >
    Save Team
  </button>
  <p
    className="save-team__text"
  >
     Search and pick 6 Pokemon to save.
    <span
      data-test="save-team-current-num"
    >
      Currently 
      6
      /6.
    </span>
  </p>
</div>
`;
